### ISSUE TITLE
[Feat] Add a plot_topics method to `GapEncoder`

### ISSUE THREAD

**Issue by Vincent-Maladiere** [skrub-data/skrub#1124]
### Problem Description

One of the `GapEncoder` 's main features is its interpretability. The documentation shows how to plot a heatmap representing activations across topics and their content, but getting this representation right is time-consuming.

### Feature Description

I suggest we bring this method directly into the `GapEncoder` class.

### Alternative Solutions

Creating a separate function elsewhere, akin to [sklearn.tree.plot_tree](https://scikit-learn.org/1.5/modules/generated/sklearn.tree.plot_tree.html), but at the cost of being less practical and visible.

**Comment by jeromedockes** [comment]
I agree this would be useful; see also https://github.com/skrub-data/skrub/discussions/909#discussioncomment-9571065

**Comment by GaelVaroquaux** [comment]
Yup!

But can we have a generic methods (ie get_report()), which could be used everywhere in skrub?

**Comment by Vincent-Maladiere** [comment]
> But can we have a generic methods (ie get_report()), which could be used everywhere in skrub?

Can you elaborate a bit on this @GaelVaroquaux? Do you see `get_report` also working for `TextEncoder` and `MinHashEncoder`?

**Comment by GaelVaroquaux** [comment]
> Can you elaborate a bit on this @GaelVaroquaux? Do you see get_report also working for TextEncoder and MinHashEncoder?

Maybe. But to me, it's more a question of something homogeneous in the library: we could have this on the Recipe (or whatever, I don't know).

I would love it if people would have a one-stop command to ask an object: give me a useful visualization of what you have in your belly. Something a bit more powerful than the default repr.

**Comment by Vincent-Maladiere** [comment]
Very interesting! Could we list somewhere the kind of visualizations we would like to see for some objects?

**Comment by jeromedockes** [comment]
> Very interesting! Could we list somewhere the kind of visualizations we would like to see for some objects?

yes let's make a list and do a sprint :)

**Comment by rcap107** [comment]
It's been more than a year since the latest update on this issue. Should we close this? Note that in the documentation and in teaching materials the GapEncoder is still described as "useful if you need interpretability", which is what this issue is about.

**Comment by GaelVaroquaux** [comment]
Let's close it. We'll see how the package evolves if we want to reconsider this. I'm not convinced that it will happen

**Comment by rcap107** [comment]
Sounds good to me
